### PR TITLE
Bump dependencies for Code Climate engine

### PR DESCRIPTION
**Pull request by pointlessone**
* Bumped Ruby to 2.3.3 (was 2.2.2p95)
* Bumped all gems to their latest versions

### CODE REVIEW

**Comment by mvz** [comment]
Looks good. @troessner do you know how we can check that this works?

File: Dockerfile [hunk 1]
```diff
@@ -13,16 +13,17 @@ ENV user app
...
+RUN adduser -u 9000 -D ${user}
+USER ${user}
+
 ADD . ${app_dir}
 
 WORKDIR ${app_dir}
 
 RUN bundle install --without debugging development
-RUN adduser -u 9000 -D ${user}
-RUN chown -R ${user}:${user} ${app_dir}
```
Comment (troessner) [inline Dockerfile]: Why is this not needed anymore? Or wasn't it needed in the first place?

File: Dockerfile [hunk 2]
```diff
@@ -13,16 +13,17 @@ ENV user app
...
+#RUN chown -R ${user}:${user} ${app_dir}
```
Comment (troessner) [inline Dockerfile]: No commented out source code please, you can just remove it.

**Comment by troessner** [comment]
>> Looks good. @troessner do you know how we can check that this works?

We can build the image and then run it via CC's cli. I can look into that but I'd first like to sort my questions ;)

File: Dockerfile (hunk 1 above)
Comment (pointlessone) [inline Dockerfile]: It is needed. I was experimenting with something and forgot to uncomment it. Will push a fix shortly.

Comment (pointlessone) [inline Dockerfile]: Updated.

**Comment by pointlessone** [comment]
> We can build the image and then run it via CC's cli. I can look into that but I'd first like to sort my questions ;)

That's how I test it.

**Comment by troessner** [comment]
Regarding 

>> Bumped all gems to their latest versions

But that is not reflected in this pull request, is it?

**Comment by pointlessone** [comment]
No, not really. There's no `Gemfile.lock` to update. The latest versions of gems will be installed on each image build. I meant to say that Gemfile and gemspec allow installation of latest version and do not need to be updated.

**Comment by troessner** [comment]
Looks good to me, merged, thanks!

**Pull request merged**